mycpp/gc_builtins.cc

```diff
@@ -243,6 +249,10 @@ bool str_equals(Str* left, Str* right) {
     return true;
   }
 
+  if (left == nullptr || right == nullptr) {
```

Comment:
OK I guess the problem here is that we don't always call `maybe_str_equals()`

This is related to the null safety issue

If we're not Null safe, then arguably we don't need both methods.  I'll record that as  a code smell